dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/compilation/VisitorTso.java

```diff
@@ -24,49 +24,45 @@ protected VisitorTso(boolean forceStart) {
                 super(forceStart);
         }
 
-	@Override
-	public List<Event> visitCreate(Create e) {
-        Store store = newStore(e.getAddress(), e.getMemValue(), e.getMo());
-        store.addFilters(C11.PTHREAD);
-        
-        return eventSequence(
-                store,
-                X86.newMemoryFence()
-        );
-	}
+        @Override
+        public List<Event> visitCreate(Create e) {
+                Store store = newStore(e.getAddress(), e.getMemValue(), "");
+                store.addFilters(C11.PTHREAD);
 
-	@Override
-	public List<Event> visitEnd(End e) {
-        return eventSequence(
-        		newStore(e.getAddress(), IValue.ZERO, e.getMo()),
-                X86.newMemoryFence()
-        );
-	}
+                return eventSequence(
+                                store,
+                                X86.newMemoryFence());
+        }
 
-	@Override
-	public List<Event> visitJoin(Join e) {
-        Register resultRegister = e.getResultRegister();
-	Local load = newLocal(resultRegister, e.getExpr());
-        load.addFilters(C11.PTHREAD);
-        
-        return eventSequence(
-        		load,
-        		newJumpUnless(new Atom(resultRegister, EQ, IValue.ZERO), (Label) e.getThread().getExit())
-        );
-	}
+        @Override
+        public List<Event> visitEnd(End e) {
+                return eventSequence(
+                                newStore(e.getAddress(), IValue.ZERO, ""),
+                                X86.newMemoryFence());
```

Comment:
Aren't TSO stores always release by default (and loads are acquire). If so, you don't need any fences here (neither in `visitCreate`).